### PR TITLE
Explosives disassembly

### PR DESCRIPTION
**Pull request by Brian-Otten**
#### Summary
```SUMMARY: Content "Lets the player disassemble explosives'"```

#### Purpose of change
As of now you can disassemble only a very small portion of launcher ammo/ explosives, and when you can it gives you pure rdx crystals, which is unrealistic and is also currently only used in making the 60L RDX barrel explosive.

#### Describe the solution
Added composition-b. This heat castable stabilized explosive and its variants are found in most military explosive munitions. Crafting items using composition B will require a forge and a crucible to recast the explosive.

Added impact and delay fuzes. these are needed to set off stabilized explosives and are found in rockets and grenades respectively.

Added disassembly of all launcher ammo (that i found. it's not impossible i missed one)

Added disassembly of the 120mm tank round.

Added the first new recipe using composition-b, an alternate method to make a pipebomb.

Corrected some weight values that had incongruities with real life.

Some ammo types move out of ammo.json and into their own files.

#### Describe alternatives you've considered
I wasn't sure how i should handle explosives in mods, mostly blazemod, because they already have their own methods of crafting and disassembly tables. they have been left untouched, but i could give them the same treatment.

I wasn't sure if i should give composition-b and the fuzes a "look-like" flag so haven't yet, would like feedback.

Currently disassembling most explosives takes fabrication 6, it could perhaps be lowered but i think this might be about right for having the confidence to take a hacksaw to a warhead.
#### Testing
Loaded up the game, disassembled all the added ammo, used the items i got from that to make some pipe bombs, and used them against some zombies.

#### Additional context
Sources used when determining disassembly results:

http://gunrf.ru/rg_granatomet_ps-7_PG-7WL_eng.html
https://en.wikipedia.org/wiki/PG-7VR
http://gunrf.ru/rg_granatomet_ps-7_TBG-7V_eng.html
http://roe.ru/eng/catalog/land-forces/strelkovoe-oruzhie/grenade-launchers/og-7v/
https://books.google.nl/books?id=K7XOZEt5gDQC&pg=SA2-PA21&lpg=SA2-PA21&dq=1.3+pounds+of+TPA&source=bl&ots=58qdsoj9Yl&sig=ACfU3U1OLrhdow30_BWIaGKN31Pie40VJw&hl=nl&sa=X&ved=2ahUKEwi0qsve7IHnAhVJL1AKHe3GAioQ6AEwAHoECAYQAQ#v=onepage&q=1.3%20pounds%20of%20TPA&f=false
https://www.defence.gov.au/UXO/_Master/docs/Types/Rocket84mmSRAAW-Rev01.pdf
https://fas.org/man/dod-101/sys/land/m830a1.htm
http://bulletpicker.com/cartridge_-120mm-heat-mp-t_-m2.html
https://cdn2.hubspot.net/hubfs/208537/assets/PDF/Direct%20Fire%20Munitions%20Sheets/120mmTankM830%20OK.pdf

When i could not find exact values for amount of material in the explosives i made best effort estimates comparing to similar rockets and to the own rockets weight/size/purpose.
Where CDDA did not have the materials used in the rocket i either used closest analogue (http://cdda-trunk.chezzo.com/incendiary), or omitted the materials for now (military solid rocket fuel)

Further changes to the disassembly results should be much easier to work on with this PR as a base.

image

Any maintainers please feel free to edit the file as you see fit.

### CODE REVIEW

**Comment by 1n17** [comment]
> Crafting items using composition B will require a forge and a crucible to recast the explosive.

Considering 215-240C melting point range for CompB that seems a bit excessive, perhaps plastic mold and surface heat would be more appropriate.

> Where CDDA did not have the materials used in the rocket i either used closest analogue (http://cdda-trunk.chezzo.com/incendiary), or omitted the materials for now (military solid rocket fuel)

Closest analogue of military solid rocket fuel would also be `incendiary` (mixture of ammonium perchlorate and aluminium/magnesium powder).

**Comment by Brian-Otten** [comment]
I considered letting people use surface heat, but my internal argument against it was that some of the tools use pretty much direct flames, and the player would really not want to risk localized hotspots. so you use the forge to get the temp just right, melt the whole thing, then cast it. This is not writ in stone, if we can think of a way to have the survivor reliably use his mess kit that would be fine by me too.

**Comment by 1n17** [comment]
Direct flame is a no go without an oil/sand batch, and personally I'd use batch with electric as well.
For melting vessel itself kitchenware pots are fine, but you do rise good point - normal mold might be cutting it bit too close (on top of it being magical, shape-shifting abstract item), molding into sand/clay would be more appropriate.

**Comment by Brian-Otten** [comment]
image
The new pipe bomb recipe. Skilled survivors can now use the large amount of explosive they might gather from something like a 120mm tank round and use it to make a large number of gunpowder primed pipe bombs.

If this PR merges (or if it's decided we want to add in this PR) my next project will be adding 3-5 more recipes the player could use their new supply of composition-b in, some of them using the new fuzes.

**Comment by 1n17** [comment]
You're doing god's work with this stuff. Have some breaching charges anywhere in your sights? Having to use whole C4 brick to blast metal doors was the reason I ever put explosives on my list (never got to it, but that's a different story).

**Comment by Brian-Otten** [comment]
Stuff i was thinking of for the future:
Homemade version of the c4 using a bunch of compound-b and a fuze with some kind of timer.
alternate craft/version of the explosive crossbow bolt.
shaped charge that functions more like an instant jackhammer (this might be tricky coding to make it feel right/realistic)
Larger improvised "grenade" with a large explosive charge and a lot of shrapnel. think 3-5 kilos.
Adding more rockets to the improvised rocket launcher that use the new parts (maybe just converting the old "unrealistic" improvised rockets)

**Comment by 1n17** [comment]
> shaped charge that functions more like an instant jackhammer (this might be tricky coding to make it feel right/realistic)

That's what I meant by breaching charge, I reckon it would just require creating iuse interface for activating and placing charge in the wall/doors and picking the right explosion power. Proper shaped charge would be a lot of fiddling with explosion code and interfaces to direct and scale the blast.

File: data/json/items/generic.json [hunk 1]
```diff
@@ -826,6 +826,34 @@
...
     "volume": 0
   },
+  {
+    "type": "GENERIC",
+    "id": "delay_fuze",
+    "symbol": ",",
+    "color": "light_gray",
+    "name": "delayed fuze",
+    "category": "spare_parts",
+    "description": "A complex mechanical fuze.  It seems it can be used to detonate stable explosives after a short time by triggering it",
```
Comment (Night-Pryanik) [inline data/json/items/generic.json]: Period in the end.

Comment (Brian-Otten) [reply data/json/items/generic.json]: Got it, thanks pryanik. will go double check elsewhere for periods at the end of sentence.

**Comment by Brian-Otten** [comment]
Thanks for the heads up, had used the materials i saw in the 40x53mm for the 40x46, will use that data and correct.
What tools should be used for disassembly of the grenades?

**Comment by anothersimulacrum** [comment]
> Crafting items using composition B will require a forge and a crucible to recast the explosive.>

You should make a requirement for this and use it, it will make this much easier to maintain.

**Comment by Brian-Otten** [comment]
Could you contact me and let me know how that works? If you mean what i think you mean that would be ideal, but not sure if/how that's possible.

**Comment by Brian-Otten** [comment]
Added the explosives_casting_standard and made the new recipe use.

**Comment by Brian-Otten** [comment]
Resolved a very simple merge conflict resulting from #36999

**Comment by mkikt4743** [comment]
it's spelled fuse not fuze, also should be using fine cutting, fine screw, fine bolt turning, etc, maybe fine metal sawing

**Comment by Brian-Otten** [comment]
https://en.wikipedia.org/wiki/Fuze

A fuse is a burning piece of rope that will ignite something, while a fuze is a mechanical/electronic detonator that will trigger under certain conditions.

I don't think the survivor would need all the fine tools to simply tear the round apart and pull out the raw material, but i might be overlooking something.

**Comment by Brian-Otten** [comment]
Fixed merge conflicts and gave this a once over. this should be good to go to merge in 0.E experimental. I'll try to contact tonkatsu to resolve the requested changes, they've been adjusted according to his suggestions.

**Comment by tenmillimaster** [comment]
Brian and I discussed a number of issues with the explosives dissassembly of 40mm projectiles; This should get re-visited when we have more reloading infrastructure in place.

**Pull request merged**